.github/workflows/build_centos7.yml

```diff
@@ -262,6 +252,119 @@ jobs:
           name: user-guide
           path: ${{ steps.create-user-guide.outputs.pdf-path }}
 
+  build_running_image:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and export
+        uses: docker/build-push-action@v6
+        with:
+          tags: centos:run
+          file: docker/centos7_test_run
+          outputs: type=docker,dest=/tmp/centos.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: centos
+          path: /tmp/centos.tar
+
+  test_assets:
+    runs-on: ubuntu-latest
+    needs: [ build, build_running_image ]
+    steps:
+      - name: Download build assets
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{needs.build.outputs.TGZ_NAME}}
+
+      - name: Download image
+        uses: actions/download-artifact@v4
+        with:
+          name: centos
+          path: /tmp
+
+      - name: Load image
+        run: |
+          docker load --input /tmp/centos.tar
+
+      - name: setup
+        run: |
+          tar --strip-components=1 -xzf ${{needs.build.outputs.TGZ_NAME}}
+
+      - uses: addnab/docker-run-action@v3
+        name: Run tests
+        with:
+          image: centos:run
+          shell: bash
+          options: -v ${{ github.workspace }}:/work
+          run: |
+            export PATH=$PATH:/usr/lib64/openmpi/bin
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
+            cd /work
+            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates --allow-run-as-root
+
+      - uses: addnab/docker-run-action@v3
+        name: run tests with 2 processes
+        with:
+          image: centos:run
+          shell: bash
+          options: -v ${{ github.workspace }}:/work
+          run: |
+            export PATH=$PATH:/usr/lib64/openmpi/bin
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
+            cd /work
+            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates -n 2 --allow-run-as-root
```

Comment:
You could merge these two steps into one

Reply:
If you're talking about the two steps  _Run tests_ and run _tests with 2 processes_ it is to better distinguish what failed in case of an error. One, the other, or both

If you're talking about the two jobs _test_assets_ and _test_single_file_asset_ same things but also jobs are run in parallel